primitives/core/src/assertion/mod.rs

```diff
@@ -135,6 +136,9 @@ pub enum Assertion {
 
 	#[codec(index = 26)]
 	NftHolder(Web3NftType),
+
+	#[codec(index = 27)]
+	Dynamic(H160) // smart contract code identifier
```

Comment:
Shall we have a list of pre-deployed contracts somewhere as comment?

Reply:
There are no pre-deployed contracts. 
We should decouple it from parachain and let product apps prepare envs for use (deploy assertions if needed)